diff --git a/ccloader/js/ccloader.js b/ccloader/js/ccloader.js
index c37133c..b38d223 100644
--- a/ccloader/js/ccloader.js
+++ b/ccloader/js/ccloader.js
@@ -23,6 +23,35 @@ export class ModLoader {
 		this.versions = {};
 		/** @type {string[]} */
 		this.extensions = this.filemanager.getExtensions();
+
+
+		const fs = require('fs')
+		const util = require('util')
+
+		const logFilePath = require('path').dirname(process.execPath) + '/log.txt'
+		const stream = fs.createWriteStream(logFilePath, { flags: 'w' })
+
+		const logFileWrite = (data) => {
+			stream.write(util.format(data) + '\n')
+		}
+
+		const logorig = console.log
+		console.log = (...args) => {
+			logorig(...args)
+			logFileWrite(args)
+		}
+
+		const warnorig = console.warn
+		console.warn = (...args) => {
+			warnorig(...args)
+			logFileWrite(args)
+		}
+
+		const errororig = console.error
+		console.error = (...args) => {
+			errororig(...args)
+			logFileWrite(args)
+		}
 	}
 
 	/**
